packages/shylu/shylu_dd/frosch/src/SchwarzOperators/FROSch_HarmonicCoarseOperator_def.hpp

```diff
@@ -0,0 +1,263 @@
+        // Build repeatedMap for the local saddle point problem // Todo:: Eigentlich gehört das in Initialize
+        // !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+        MapPtr repeatedMap = assembleRepeatedMap();
+        
+        // Build local saddle point problem
+        CrsMatrixPtr repeatedMatrix = FROSch::ExtractLocalSubdomainMatrix(this->K_,repeatedMap);
+        
+        // Extract submatrices
+        GOVec indicesGammaDofsAll(0);
+        GOVec indicesIDofsAll(0);
+        LO tmp = 0;
+        
+        for (UN i=0; i<NumberOfBlocks_; i++) {
+            for (UN j=0; j<IndicesGammaDofs_[i].size(); j++) {
+                indicesGammaDofsAll.push_back(tmp+IndicesGammaDofs_[i][j]);
+            }
+            for (UN j=0; j<IndicesIDofs_[i].size(); j++) {
+                indicesIDofsAll.push_back(tmp+IndicesIDofs_[i][j]);
+            }
+            tmp += IndicesGammaDofs_[i].size()+IndicesIDofs_[i].size(); // Ist das mit tmp korrekt?
+        }
+        
+        CrsMatrixPtr kII;
+        CrsMatrixPtr kIGamma;
+        CrsMatrixPtr kGammaI;
+        CrsMatrixPtr kGammaGamma;
+        
+        FROSch::BuildSubmatrices(repeatedMatrix,indicesIDofsAll(),kII,kIGamma,kGammaI,kGammaGamma);
+        //if (this->Verbose_) std::cout << kII->NumMyRows() << " " << kGammaI->NumMyRows();
+        //if (this->Verbose_) std::cout << *kII << *kIGamma << *kGammaI << *kGammaGamma;
+        //FROSCH_ASSERT(0!=0,"STOP!");
+        
+        // Assemble coarse map
+        MapPtr coarseMap = assembleCoarseMap(); // Todo:: Eigentlich gehört das in Initialize
+        // !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

Comment:
May be not key board error ?

Reply:
This is just a very strong reminder for myself. I am going to continue restructuring the CoarseOperators and there are two maps that should be build in initialize() instead of compute().